File: frontend-angular/src/app/models/Habilidad.ts
export class Habilidad {
  protected _id: number = -1;
  protected _nombre: string = '';
  protected _color: string = '';
  protected _clase: string = '';
  protected _fechaI: Date = new Date();

  public constructor() {}

  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }

  public get nombre(): string {
    return this._nombre;
  }
  public set nombre(value: string) {
    this._nombre = value;
  }

  public get color(): string {
    return this._color;
  }
  public set color(value: string) {
    this._color = value;
  }

  public get clase(): string {
    return this._clase;
  }
  public set clase(value: string) {
    this._clase = value;
  }

  public get fechaI(): Date {
    return this._fechaI;
  }
  public set fechaI(value: Date) {
    this._fechaI = value;
  }
}

File: backend-spring-boot/src/main/java/com/facundocastro/api/model/Habilidad.java
package com.facundocastro.api.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.sun.istack.NotNull;
import lombok.Data;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;
import java.util.Date;

@Data
@Entity
public class Habilidad {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    private String nombre;
    @NotNull
    private String color;
    @NotNull
    private String clase;
    private Date fechaI;
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JsonIgnore
    private Usuario usuario;
}
